src/DynamoPackages/PackageLoader.cs

```diff
@@ -85,6 +85,11 @@ public string DefaultPackagesDirectory
             get { return defaultPackagesDirectoryIndex != -1 ? packagesDirectories[defaultPackagesDirectoryIndex] : null; }
         }
 
+        internal void SetPackagesDownloadDirectory(string downloadDirectory)
+        {
+            defaultPackagesDirectoryIndex = packagesDirectories.IndexOf(downloadDirectory);
```

Comment:
This does not work right now as I discovered that for some reason there is a mismatch in the `packagesDirectories` list maintained by the `PackageLoader` and the `CustomPackageFolders` held in the `PreferenceSettings`. For example, the former has the entry `C:\Users\pratapa\AppData\Roaming\Dynamo\Dynamo Core\2.12\packages`, while the latter has `C:\Users\pratapa\AppData\Roaming\Dynamo\Dynamo Core\2.12`, without the `packages` subdirectory.

Reply:
@aparajit-pratap this reminds me of an issue that RSA devs reported to us. I can't find the slack thread at the moment but I think we were both confused. Seems like a bug rather than something intentional.

Reply:
@mjkkirschner I think this is the one you're referring to: https://autodesk.slack.com/archives/CTACCQ91R/p1596438758024600